src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/jpa/repository/ReasonTypeRepositoryTest.kt

```diff
@@ -31,19 +34,22 @@ class ReasonTypeRepositoryTest : TestBase() {
 
   @Test
   fun getReasonTypeByCode() {
-    var reason = repository.findById("VIC").orElseThrow { Exception("ReasonType not found") }
+    val reason = repository.findById("VIC").orElseThrow { Exception("ReasonType not found") }
     assertThat(reason).isNotNull
     assertThat(reason.description).isEqualTo("Victim")
   }
 
   @Test
   fun updateRecord() {
-    var reason = repository.findById("VIC").orElseThrow { Exception("ReasonType not found") }
-    var whenCreated = reason.whenCreated
-
+    val reason = repository.findById("VIC").orElseThrow { Exception("ReasonType not found") }
+    val whenCreated = reason.whenCreated
     repository.save(reason.copy(description = "UPDATED"))
 
-    var reasonUpdated = repository.findById("VIC").orElseThrow { Exception("ReasonType not found") }
+    TestTransaction.flagForCommit()
+    TestTransaction.end()
+    TestTransaction.start()
```

Comment:
How does this work? Does it mean that by default a `@Test` method is wrapped in a transaction or something?

And if that's the case how does reading after a save in a test work? 🤔  Sorry too many stupid Spring Boot questions in these PRs

Reply:
Yes `@JpaDataTest` sets up a transaction and allows tx to be commited.  JPA doesn't actually issue SQL unless the transaction is committed.  So without this bit you aren't actually issuing an INSERT/UPDATE